core/src/Streamly/Internal/Data/Stream/Type.hs

```diff
@@ -173,7 +199,396 @@ instance Monad m => Applicative (Stream m) where
     -- (<*)  = K.apSerialDiscardSnd
 
 MONAD_COMMON_INSTANCES(Stream,)
-LIST_INSTANCES(Stream)
 NFDATA1_INSTANCE(Stream)
-FOLDABLE_INSTANCE(Stream)
-TRAVERSABLE_INSTANCE(Stream)
+
+-- XXX Renamed to foldWith because Stream has a Foldable instance having
+-- method fold.
+--
+-- | Fold a stream using the supplied left 'Fold' and reducing the resulting
+-- expression strictly at each step. The behavior is similar to 'foldl''. A
+-- 'Fold' can terminate early without consuming the full stream. See the
+-- documentation of individual 'Fold's for termination behavior.
+--
+-- >>> Stream.foldWith Fold.sum (Stream.enumerateFromTo 1 100)
+-- 5050
+--
+-- Folds never fail, therefore, they produce a default value even when no input
+-- is provided. It means we can always fold an empty stream and get a valid
+-- result.  For example:
+--
+-- >>> Stream.foldWith Fold.sum Stream.nil
+-- 0
+--
+-- However, 'foldMany' on an empty stream results in an empty stream.
+-- Therefore, @Stream.foldWith f@ is not the same as @Stream.head . Stream.foldMany
+-- f@.
+--
+-- @foldWith f = Stream.parse (Parser.fromFold f)@
+--
+-- /Pre-release/
+{-# INLINE foldWith #-}
+foldWith :: Monad m => Fold m a b -> Stream m a -> m b
+foldWith fld m = D.fold fld $ toStreamD m
+
+-- XXX Renamed to "list" because fromList is present in IsList instance.
+--
+-- |
+-- @
+-- fromList = 'Prelude.foldr' 'K.cons' 'K.nil'
+-- @
+--
+-- Construct a stream from a list of pure values. This is more efficient than
+-- 'K.fromFoldable' for serial streams.
+--
+-- @since 0.4.0
+{-# INLINE_EARLY list #-}
+list :: Monad m => [a] -> Stream m a
+list = fromStreamD . D.fromList
+
+------------------------------------------------------------------------------
+-- Conversion to and from direct style stream
+------------------------------------------------------------------------------
+
+-- These definitions are dependent on what is imported as S
+{-# INLINE fromStreamS #-}
+fromStreamS :: (Monad m) => S.Stream m a -> Stream m a
+fromStreamS = Stream . S.toStreamK
+
+{-# INLINE toStreamS #-}
+toStreamS :: (Monad m) => Stream m a -> S.Stream m a
+toStreamS = S.fromStreamK . getStream
```

Comment:
We should remove these. Instead the toStreamK/fromStreamK functions - that have been removed in this change - should be used.